### PR TITLE
Scoreboard added

### PR DESCRIPTION
**Pull request by AlexBrin**
## Introduction
Scoreboard added

### API changes

* `Server->getScoreboardManager`
* `ScoreboardManager->create(Player $player, ?string $title = null)`
* `ScoreboardManager->destroy(Player $player)`
* `ScoreboardManager->setScoreboardClass(string $newScoreboard)`
* `Scoreboard->create(string $title, bool $oldRemove = false)`
* `Scoreboard->destroy()`
* `Scoreboard->setLines(string[] $lines)`
* `Player->addScoreboard(string title, array $lines = [])`
* `Player->removeScoreboard()`

### Send scoreboard:
```php
$player->sendScoreboard("Awesome scoreboard", [
  "My scoreboard",
  "Veeery long scoreboard line",
]);
```

### Custom scoreboard:
```php
class MyScoreboard extends \pocketmine\scoreboard\Scoreboard {
// todo
}
```
And install a new instance:
```php
$server->getScoreboardManager()->setScoreboardClass(MyScoreboard::class);
```

### CODE REVIEW

**Comment by Encritary** [comment]
There's already a scoreboard PR and, don't get me wrongly, IMHO it's much better.

File: src/pocketmine/scoreboard/ScoreboardManager.php [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+	 * @return string
+	 */
+	public function getScoreboardClass(): string {
+		return $this->currentClass;
+	}
+
+	/**
+	 * @param $class
+	 */
+	public function setScoreboardClass($class) {
```
Comment (SOF3) [inline src/pocketmine/scoreboard/ScoreboardManager.php]: I hope you aren't trying to make another PlayerCreationEvent...

**Comment by SOF3** [comment]
This looks more like a statically-linked library than an API. I don't see any evidence of attempts to maintain plugin compatibility.

File: src/pocketmine/scoreboard/ScoreboardManager.php (hunk 1 above)
Comment (AlexBrin) [reply src/pocketmine/scoreboard/ScoreboardManager.php]: Of course not. May be necessary to use your own scoreboard.
E.g. to change the line numbering.

Comment (SOF3) [reply src/pocketmine/scoreboard/ScoreboardManager.php]: ScoreboardManager is a singleton. This is generating the same problem as PlayerCreationEvent.

**Comment by AlexBrin** [comment]
@SOF3, what compatibility are we talking about? 
Yes, it statically-linked, but only to avoid problems when working with the scoreboard. E.g., from different plugins. But it is, nevertheless, API.

File: src/pocketmine/scoreboard/ScoreboardManager.php [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+	 * @param Player $player
+	 * @param string|null $title
+	 * @return Scoreboard
+	 */
+	public function create(Player $player, ?string $title = null): Scoreboard {
+		if(isset($this->scoreboards[$player->getName()])) {
+			$scoreboard = $this->scoreboards[$player->getName()];
+		} else {
+			$scoreboard = new $this->currentClass($player);
+			$this->scoreboards[$player->getName()] =& $scoreboard;
```
Comment (dktapps) [inline src/pocketmine/scoreboard/ScoreboardManager.php]: Objects are always assigned by reference. This reference-to-reference makes zero sense.

File: src/pocketmine/scoreboard/ScoreboardManager.php (hunk 1 above)
Comment (dktapps) [reply src/pocketmine/scoreboard/ScoreboardManager.php]: This entire class is pointless. It's just a reduction of flexibility.

**Pull request closed without merging**